node.go

```diff
@@ -37,73 +37,159 @@ type Node[T any] struct {
 	// prefix is the common prefix we ignore
 	prefix []byte
 
-	// Edges should be stored in-order for iteration.
-	// We avoid a fully materialized slice to save memory,
-	// since in most cases we expect to be sparse
-	edges edges[T]
+	// bitmap represents which edges exist.
+	// There are 256 possible edges (one per byte),
+	// so we use 4 uint64s for 256 bits total.
+	bitmap [4]uint64
+	edges  []*Node[T]
 }
 
-func (n *Node[T]) isLeaf() bool {
-	return n.leaf != nil
+// setBit sets the bit for a given label
+func setBit(bitmap *[4]uint64, label byte) {
+	block := label >> 6
+	bitPos := label & 63
+	bitmap[block] |= 1 << bitPos
 }
 
-func (n *Node[T]) addEdge(e edge[T]) {
-	num := len(n.edges)
-	idx := sort.Search(num, func(i int) bool {
-		return n.edges[i].label >= e.label
-	})
-	n.edges = append(n.edges, e)
-	if idx != num {
-		copy(n.edges[idx+1:], n.edges[idx:num])
-		n.edges[idx] = e
+// clearBit clears the bit for a given label
+func clearBit(bitmap *[4]uint64, label byte) {
+	block := label >> 6
+	bitPos := label & 63
+	mask := uint64(1) << bitPos
+	bitmap[block] &^= mask
+}
+
+// bitSet checks if bit for label is set
+func bitSet(bitmap [4]uint64, label byte) bool {
+	block := label >> 6
+	bitPos := label & 63
+	return (bitmap[block] & (1 << bitPos)) != 0
```

Comment:
I would document what each of these steps does very clearly since most folks aren't doing bit manipulation day in and day out so it will be much easier for future readers to follow along with documentation. Example.

```
 	// Determine which block the label falls into by shifting 6 bits to the right. 
 	// This is the equivalent of dividing by 64.
 	block := label >> 6
       
 	// Since the block captures the two high order bits, we do an AND with the lower 6 bits (0x00111111 == 63)
        // to determine which bit to check within the selected block.
 	bitPos := label & 63
 	return (bitmap[block] & (1 << bitPos)) != 0
```

This especially applies to the more complex methods below such as `findInsertionIndex`

Reply:
Makes sense, will update